### ISSUE TITLE
enrichDF2enrichResult only works when Term IDs are a factor

### ISSUE THREAD

**Issue by john-lee-johnson** [jmw86069/multienrichjam#3]
Hello,

I've noticed the `enrichDF2enrichResult` function only works when the Term IDs are a factor and not as a character when I go to later use the `cnetplot` function in the enrichplot package.

When I pass a DF to `enrichDF2enrichResult` with Term IDs as a character and try to plot my results with `cnetplot` I get this error: 
      ``` Error in graph.data.frame(x, directed = FALSE) : 
       the data frame should contain at least two columns```

However, it works when I pass the same DF to `enrichDF2enrichResult` with Term IDs as a factor and then use `cnetplot`

I have clusterProfiler v3.16.0 and multienrichjam v0.0.35.900 installed

Wondering if you had any thoughts

I have included a reproducible example below:

```
#Create Term IDs as character
Term.ID <- c("GO:0001501", "GO:0001503", "GO:0001525", "GO:0001568", "GO:0001775", "GO:0001816", "GO:0001817", "GO:0001944", "GO:0001952", "GO:0002376")
#Also create Term IDs as factor
Term.ID.factor <- factor(Term.ID)
#List enriched genes of each Term ID
Genes <- c("COL3A1/COL5A2/FGFR1/JAG2/LUM/MSX1/POH/STC1/TIMP1/VCAN", "COL5A2/JAG1/POH/PTK2/SPP1/STC1/VCAN", "FGFR1/ITGAV/JAG1/NRP1/PDGFA/PF4/PTK2/VAV2/VEGFA", "COL3A1/FGFR1/ITGAV/JAG1/NRP1/PDGFA/PTK2/VAV2/VEGFA", "APP/COL3A1/CXCL6/ITGAV/JAG2/OLR1/PDGFA/PF4/PGLYRP1/POH/PRG2/THBD/TIMP1/TNFRSF21/VAV2", "APP/CXCL6/FGFR1/ITGAV/LPL/LUM/PF4/PGLYRP1/POSTN/TNFRSF21", "APP/CXCL6/FGFR1/ITGAV/LPL/LUM/PF4/PGLYRP1/POSTN/TNFRSF21", "COL3A1/FGFR1/ITGAV/JAG1/NRP1/PDGFA/PF4/PTK2/VAV2/VEGFA", "JAG1/NRP1/POSTN/PTK2/VEGFA", "APP/COL3A1/CXCL6/ITGAV/JAG1/JAG2/OLR1/PF4/PGLYRP1/POH/PRG2/PTK2/THBD/TNFRSF21/VAV2/VEGFA/VTN")
E.value <- c(1.25e-05, 2.01e-02, 1.11e-03, 1.34e-03, 1.24e-06, 1.11e-03, 4.53e-04, 1.02e-03, 1.62e-02, 1.25e-03)
#Term ID Description
Description <- c("skeletal system development", "ossification", "angiogenesis", "blood vessel development", "cell activation", "cytokine production", "regulation of cytokine production", "vasculature development", "regulation of cell-matrix adhesion", "immune system process")
#Create DFs
test_df <- cbind.data.frame(Term.ID, Genes, E.value, Description)
test_df_factor <- cbind.data.frame(Term.ID.factor, Genes, E.value, Description)
colnames(test_df_factor) <- c("Term.ID", "Genes", "E.value", "Description")

ego <- enrichDF2enrichResult(test_df, keyColname = "Term.ID", geneColname = "Genes", pvalueColname = "E.value", pvalueCutoff=0.05, pAdjustMethod="none", verbose = TRUE)
ego_factor <- enrichDF2enrichResult(test_df_factor, keyColname = "Term.ID", geneColname = "Genes", pvalueColname = "E.value", pvalueCutoff=0.05, pAdjustMethod="none", verbose = TRUE)

#Has error
cnetplot(ego)

#Works
cnetplot(ego_factor)
```

**Comment by jmw86069** [comment]
Thank you for posting the issue, and reproducible example! Perfect.

I was able to find the source of issue in the `enrichplot` package, specifically `enrichplot:::extract_geneSets()` function. It calls `DOSE:::geneInCategory.enrichResult()` which assigns names using `rownames(ego@result)`. Those names are expected to match the `ID` column in the enrichment table.

I implemented a fix, which is to assign rownames using values in `"ID"` column. I tested your workflow above and it worked for me. Hopefully this fixes your issue.
